api/envoy/config/listener/v3/listener.proto

```diff
@@ -339,6 +339,10 @@ message Listener {
   // provided net.core.somaxconn will be used on Linux and 128 otherwise.
   google.protobuf.UInt32Value tcp_backlog_size = 24;
 
+  // The maximum number of connections to accept per socket event. If no value is
+  // provided we will accept all acceptable connections.
+  google.protobuf.UInt32Value max_connections_to_accept_per_socket_event = 34;
```

Comment:
For someone not familiar, this is quite confusing. When would there be a connection overflow for an event?

Reply:
Assuming your terminology of "connection overflow" is `pending_connections > max_connections_to_accept_per_socket_event` then the answer is:

https://github.com/envoyproxy/envoy/pull/27773/files#diff-6f9c718224c533c13c2c0ba1d5abaab86be9d0cc73808749c77934e9f9b0d5d0R207-R211 

Which can be seen in https://github.com/envoyproxy/envoy/pull/27773/files#diff-ed5b7fa9e08e0fc30d0da2b77a3432fb99a8994d975fd768828f6f5bf482251eR197-R220

Reply:
This gives users chances to regularly bound how much work they'll do in this given call. 

/proc/sys/net/core/somaxconn of the underlying system is insufficient for bounding this as while Envoy can be draining the queue it’s possible the kernel may add additional items

Reply:
I find the description confusing, probably because it is a pretty low-level implementation detail (the number of events in a single invocation of the `onSocketEvent()` callback), that I'm not certain this is the right place for it. I'm not aware of anything similar that is done in Envoy, but if you do, please let me know.

Another comment: it is unclear what will happen to the connections that aren't accepted in this call (will they be rejected or accepted later). The description should clarify this.

Reply:
Also please add in the description that in what scenario a user should be configuring it with some example values.

Reply:
> Another comment: it is unclear what will happen to the connections that aren't accepted in this call (will they be rejected or accepted later). The description should clarify this.

+1 to this. I'm assuming the eventloop will immediately run again, and it could do another batch up to the configured size. But the behavior should be clearly documented.

Reply:
Added comments in the user description and to the proto itself. I think this is an appropriate place e.g. see `tcp_backlog_size` which is similar but not fully sufficient for this usecase also in this proto.

Example values are deployment specific but I've listed use cases.

> I'm assuming the eventloop will immediately run again, and it could do another batch up to the configured size. But the behavior should be clearly documented.

Yep your expectation is correct, documented here and description has a pointer to a test demonstrating this.